### PR TITLE
adds init_app method to HealthCheck and EnvironmentDump classes

### PR DESCRIPTION
**Pull request by frankV**
Adds support for Flask application factory. No public tests so I could only test with available methods.

### CODE REVIEW

**Comment by frankV** [comment]
if you let me know what happened I can try to fix and re-submit.

**Comment by RyanGWU82** [comment]
Hi Frank, I think the problem is with our internal (private) Jenkins server—it doesn't work correctly on third party forks. Nothing to worry about. The Jenkins server does pep8 and pylint validation on the code, but otherwise I don't think there are any unit tests for this project. We can always fix those after it's merged, if that's necessary.

**Comment by frankV** [comment]
think init_app is breaking add_check methods, I'm getting None for app when I try to add them.
Do you guys happen to have an example for SQLAlchemy, or a similar extension. I think I'm missing something when trying to create a checker.

File: healthcheck/__init__.py [hunk 1]
```diff
@@ -67,6 +67,12 @@ def __init__(self, app=None, path=None, success_status=200,
         if self.app and self.path:
             app.add_url_rule(self.path, view_func=self.check, **options)
 
+    def init_app(self, app=None, path=None, **kwargs):
+        _healthcheck = HealthCheck(app, path, **kwargs)
```
Comment (frankstratton) [inline healthcheck/__init__.py]: I don't think you need to create a new HealthCheck object here; you should use `self`

**Comment by frankstratton** [comment]
closing in favor of https://github.com/Runscope/healthcheck/pull/17

**Comment by frankV** [comment]
Ah man, so sorry. I forgot about this.

**Comment by frankstratton** [comment]
No worries; I appreciate the time you put in.

**Pull request closed without merging**